### ISSUE TITLE
Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by kavington** [jfmengels/eslint-plugin-lodash-fp#13]
Hey there, thanks for the great plugin. Am getting an error I can't seem to resolve when editing in Atom / Sublime Text.

Using Node 6

The error
```
Error: Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at isRequireOfLodashSubModule (/Users/kav/Sites/playground/animplay/node_modules/eslint-plugin-lodash-fp/rules/no-submodule-destructuring.js:16:21)
    at Object.info.merge.VariableDeclarator (/Users/kav/Sites/playground/animplay/node_modules/eslint-plugin-lodash-fp/rules/no-submodule-destructuring.js:34:52)
    at EventEmitter.predefinedRules.(anonymous function) (/Users/kav/Sites/playground/animplay/node_modules/eslint-plugin-lodash-fp/rules/core/enhance.js:86:34)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.enterNode (/Users/kav/Sites/playground/animplay/node_modules/eslint/lib/util/node-event-generator.js:40:22)
    at CodePathAnalyzer.enterNode (/Users/kav/Sites/playground/animplay/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:607:23)
    at CommentEventGenerator.enterNode (/Users/kav/Sites/playground/animplay/node_modules/eslint/lib/util/comment-event-generator.js:97:23)
    at Controller.traverser.traverse.enter (/Users/kav/Sites/playground/animplay/node_modules/eslint/lib/eslint.js:886:36)
    at Controller.__execute (/Users/kav/Sites/playground/animplay/node_modules/estraverse/estraverse.js:397:31)
```

Import in the offending file
```javascript
import { curry, find, flow, get, head, map } from 'lodash/fp';
import bodymovin from 'bodymovin';

/**
 * @type {number}
 */
const SEGMENT_LENGTH_SECONDS = 1;

...
```

eslintrc.yml
```yaml
extends: [airbnb-base]
parser: babel-eslint
parserOptions: { ecmaVersion: 7, sourceType: module }
plugins: [babel, lodash-fp]
rules:
  babel/generator-star-spacing: 1
  babel/new-cap: 1
  babel/object-shorthand: 1
  babel/no-await-in-loop: 1
  lodash-fp/consistent-compose: [error, flow]
  lodash-fp/consistent-name: [error, _]
  lodash-fp/no-argumentless-calls: error
  lodash-fp/no-chain: error
  lodash-fp/no-extraneous-function-wrapping: error
  lodash-fp/no-single-composition: error
  lodash-fp/no-submodule-destructuring: error
  lodash-fp/prefer-compact: error
  lodash-fp/prefer-composition-grouping: error
  lodash-fp/prefer-constant: [error, { arrowFunctions: false }]
  lodash-fp/prefer-flat-map: error
  lodash-fp/prefer-get: error
  lodash-fp/prefer-identity: [error, { arrowFunctions: false }]
  lodash-fp/use-fp: error
```

package.json dev dependencies
```json
"devDependencies": {
  "autoprefixer": "^6.3.6",
  "ava": "^0.14.0",
  "babel-core": "^6.7.7",
  "babel-eslint": "^6.0.4",
  "babel-loader": "^6.2.4",
  "babel-plugin-lodash": "^2.3.0",
  "babel-plugin-transform-es2015-modules-commonjs": "^6.7.7",
  "babel-plugin-transform-runtime": "^6.7.5",
  "babel-preset-es2015-webpack": "^6.4.1",
  "babel-preset-stage-0": "^6.5.0",
  "babel-register": "^6.7.2",
  "css-loader": "^0.23.1",
  "eslint": "^2.9.0",
  "eslint-config-airbnb-base": "^2.0.0",
  "eslint-plugin-babel": "^3.2.0",
  "eslint-plugin-import": "^1.6.1",
  "eslint-plugin-lodash-fp": "^1.1.1",
  "extract-text-webpack-plugin": "^1.0.1",
  "file-loader": "^0.8.5",
  "html-webpack-plugin": "^2.16.0",
  "json-loader": "^0.5.4",
  "postcss-loader": "^0.9.1",
  "promise-loader": "^1.0.0",
  "rimraf": "^2.5.2",
  "style-loader": "^0.13.1",
  "webpack": "^2.1.0-beta.6",
  "webpack-dev-server": "^2.0.0-beta"
},
```

Also tried using this eslint config:
```yaml
extends: [airbnb-base, 'plugin:lodash-fp/recommended']
parser: babel-eslint
parserOptions: { ecmaVersion: 7, sourceType: module }
plugins: [babel, lodash-fp]
rules:
  babel/generator-star-spacing: 1
  babel/new-cap: 1
  babel/object-shorthand: 1
  babel/no-await-in-loop: 1
```

Did not fix it, instead an additional error is thrown in files that linted fine before:
```
Definition for rule 'lodash-fp/prefer-identify' was not found
```

Example of an offending import:
```javascript
import { map, zipObject } from 'lodash/fp';

async function animationLoader(name) ...
```

Same things happen when I remove all rules and babel plugin from eslintrc.

Thanks heaps!

**Comment by jfmengels** [comment]
I fixed it and released it in `v1.1.2`. there were two problems:
- There was an incorrect "assumption" in the code that any variable declaration with object destructuration had a call in the right-side member. i.e. probably something like `var {foo} = bar;` n your code made it crash.
- There was a typo in the name of one of the rules (`prefer-identify` vs`prefer-identity`)

Thanks for the report!
Let me know if you have other issues!

**Comment by kavington** [comment]
Worked a treat, thank you!
